### PR TITLE
feat: add cloud-init message

### PR DESCRIPTION
**Pull request by abehnia**
Based on `livepatch` and `ubuntupro` this was created as a manager but it can be changed to monitor as well.

### CODE REVIEW

**Comment by silverdrake11** [comment]
I have not looked at your code but livepatch and ubuntu pro are manager because they need to admin access. "pro status" works on the current user like "ubuntu" in an lxd but does not work with the "landscape" user. To test whether manager is needed, try running the command with the "landscape" user, if you get an error then continue to use manager, if it works fine, then monitor would be best as it uses the least privileges

File: landscape/client/manager/cloudinit.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        data["return_code"] = -2
+        data["error"] = str(exc)
+        data["output"] = ""
+    else:
+        string_output = completed_process.stdout.strip()
+        try:
+            # INFO: We don't want to parse an empty string.
+            if string_output:
+                json_output = json.loads(string_output)
+                # INFO: Only return relevant information from cloud init.
```
Comment (silverdrake11) [inline landscape/client/manager/cloudinit.py:53]: This is a good call as there can be sensitive data in the cloud init file

File: landscape/client/manager/cloudinit.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                output["availability_zone"] = json_output.get(
+                    "availability_zone",
+                    "",
+                ) or json_output.get("availability-zone", "")
```
Comment (silverdrake11) [inline landscape/client/manager/cloudinit.py:57]: Seems like there can be dashes or underscores in the cloud init file?

File: landscape/client/manager/cloudinit.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+class CloudInit(DataWatcherManager):
+
+    message_type = "cloud-init"
+    message_key = message_type
+    scope = "cloud-init"
+    persist_name = message_type
+    run_immediately = True
+    run_interval = 3600 * 24  # 24h
```
[the remark this answers is not in the document]
Comment (silverdrake11) [reply landscape/client/manager/cloudinit.py:15]: I suppose the cloud init cannot change though? If that's the case then 24 hours is fine.

[the remark this answers is not in the document]
Comment (abehnia) [reply landscape/client/manager/cloudinit.py]: Yes, it should be ready-only.

File: landscape/client/manager/cloudinit.py (hunk 2 above)
Comment (abehnia) [reply landscape/client/manager/cloudinit.py]: Yes.

File: landscape/client/monitor/cloudinit.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    message_key = message_type
+    scope = message_type
+    persist_name = message_type
+    run_immediately = True
+    run_interval = 3600 * 24  # 24h
+
+    def get_data(self) -> str:
+        return json.dumps(get_cloud_init(), sort_keys=True)
+
+    def register(self, monitor):
```
Comment (silverdrake11) [inline landscape/client/monitor/cloudinit.py:19]: You shouldn't need this register. See https://github.com/canonical/landscape-client/pull/130/files and https://github.com/canonical/landscape-client/pull/158

**Review by silverdrake11 — CHANGES_REQUESTED** [review]

File: landscape/client/monitor/cloudinit.py (hunk 4 above)
Comment (abehnia) [reply landscape/client/monitor/cloudinit.py:19]: Went over offline.

**Review by silverdrake11 — APPROVED** [review]
LGTM

**Pull request merged**